### DISCUSSION TITLE
How to manually change url hash while using BrowserHistory

### DISCUSSION THREAD

**Discussion by ehsanghorbani190** [remix-run/remix#6683] [Q&A]
Hi! I have want to use BrowserHistory for my webapp to navigate between modules. But also I want to change the hash in URL with some user interactions. For example, if they open my site, the URL will be: `www.example.com/#my_hash`. And if they use a button or hover on something, it will become like : `www.example.com/faq/#my_second_hash`. I thought of manually modifying `window.location.hash`, but I'm not sure if that's the best way to do it and if it's stable. Any ideas?

**Comment by xHomu** [comment]
useNavigate() is what we usually use when you want to dynamically change the url. Usually, it would trigger a navigation event, but I believe hashes shouldn't. Something like

```jsx
const navigate = useNavigate();

function someIOevent(e) {
if(something happens here) navigate("#newanchor");
...
}

...
```

window.location.hash should also work, here's how Kent does something similar https://github.com/kentcdodds/kentcdodds.com/blob/main/app/utils/misc.tsx#L298
